model/feature-flag/logs.yaml

```
@@ -1,11 +1,49 @@
 groups:
-  - id: log-feature_flag
+  - id: feature_flag.evaluation
     type: event
-    stability: experimental
-    name: feature_flag
+    name: feature_flag.evaluation
     brief: >
-      This event describes feature flag evaluation representation on a Log Record.
+      This semantic convention defines the attributes used to represent a
+      feature flag evaluation as an event.
+      A `feature_flag.evaluation` event SHOULD be emitted whenever a feature flag
+      value is evaluated, which may happen many times over the course of an
+      application lifecycle.
+      For example, a website A/B testing different animations may evaluate a
+      flag each time a button is clicked.
+      A `feature_flag.evaluation` event is emitted on each evaluation even if the result is the same.
     attributes:
       - ref: feature_flag.key
-      - ref: feature_flag.provider_name
+        requirement_level: required
       - ref: feature_flag.variant
+        requirement_level:
+          conditionally_required: If feature flag provider supplies a variant or equivalent concept.
+      - ref: feature_flag.system
+        requirement_level: recommended
+      - ref: feature_flag.context.id
+        requirement_level: recommended
+      - ref: feature_flag.version
+        requirement_level: recommended
+      - ref: feature_flag.set.id
+        requirement_level: recommended
+      - ref: feature_flag.evaluation.reason
+        requirement_level: recommended
+      - ref: feature_flag.evaluation.error.code
+        requirement_level:
+          conditionally_required: Required if an error occurred during flag evaluation and `feature_flag.evaluation.error.message` is not set, recommended otherwise.
```

not sure I follow, could we elaborate? I.e. (nerdy interpretation could be):

- error occurred and `feature_flag.evaluation.error.message` is not set - required
- error, but `feature_flag.evaluation.error.message` is not set - recommended (why not required?)
- no error and `feature_flag.evaluation.error.message` is set - recommended  (should not  be set, right?)
- no error and `feature_flag.evaluation.error.message` is not  set - recommended (should not  be set, right?)

It does not seem right.

I'd simplify it to:

```suggestion
          conditionally_required: if and only if an error occurred during flag evaluation.
```

---

I was trying to make it so that either error.message or error.code was required, but not both. In some cases error code might not be available, but if it is available it is likely sufficient on its own without message